dragonfly/implementations/daq_run_interface.py

```diff
@@ -82,10 +83,20 @@ def run_name(self):
     def run_name(self, value):
         self._run_name = value
         self._acquisition_count = 0
-        result = self.provider.cmd(self.run_table_endpoint, 'do_insert', payload={'run_name':value})
-        self.run_id = result['run_id']
-        self._start_time = result['start_timestamp']
-
+        try:
+            result = self.provider.cmd(self.run_table_endpoint, 'do_insert', payload={'run_name':value})
+            self.run_id = result['run_id']
+            self._start_time = result['start_timestamp']
+        except Exception as err:
+            logger.critical('failed to insert run_name to the db, obtain run_id, and start_timestamp. error:\n{}'.format(str(err)))
+            # end the run
+            if self._stop_handle is not None:
+                self.service._connection.remove_timeout(self._stop_handle)
+                logger.critical('run <{}> was not started'.format(value))
+                self._stop_handle = None
+                self._run_name = None
+                self.run_id = None
```

Comment:
Ok so you think it would be better to move raising the exception from line 117 to (right) after line 98? Then lines 115-117 would be removed. That does seem cleaner so I would agree.